bchrpc/server.go

```diff
@@ -1113,16 +1208,421 @@ func (s *GrpcServer) GetMerkleProof(ctx context.Context, req *pb.GetMerkleProofR
 	return resp, nil
 }
 
+// GetTokenMetadata returns metadata associated with a Token ID
+func (s *GrpcServer) GetTokenMetadata(ctx context.Context, req *pb.GetTokenMetadataRequest) (*pb.GetTokenMetadataResponse, error) {
+	if s.slpIndex == nil {
+		return nil, status.Error(codes.Unavailable, "slpindex required")
+	}
+
+	tokenMetadata := make([]*pb.TokenMetadata, 0)
+	for _, hash := range req.GetTokenIds() {
+		tokenID, err := chainhash.NewHash(hash)
+		if err != nil {
+			return nil, status.Error(codes.Aborted, fmt.Sprintf("token ID hash %s is invalid %s", hex.EncodeToString(hash), err.Error()))
+		}
+
+		tm, err := s.buildTokenMetadata(*tokenID)
+		if err != nil {
+			return nil, status.Error(codes.Aborted, "token ID "+hex.EncodeToString(hash)+" does not exist")
+		}
+
+		tokenMetadata = append(tokenMetadata, tm)
+	}
+
+	resp := &pb.GetTokenMetadataResponse{
+		TokenMetadata: tokenMetadata,
+	}
+
+	return resp, nil
+}
+
+// GetParsedSlpScript returns a parsed object from a provided serialized slp OP_RETURN message
+func (s *GrpcServer) GetParsedSlpScript(ctx context.Context, req *pb.GetParsedSlpScriptRequest) (*pb.GetParsedSlpScriptResponse, error) {
+	resp := &pb.GetParsedSlpScriptResponse{}
+	slpMsg, er := v1parser.ParseSLP(req.GetSlpOpreturnScript())
+	if er != nil {
+		resp.ParsingError = er.Error()
+		return resp, nil
+	}
+	if slpMsg.TokenType == 0x41 && slpMsg.TransactionType == "GENESIS" {
```

Comment:
Should we have a constant that represents this hard coded value? `0x41` is not going to be commonly known.

Reply:
done.